main.go

```diff
@@ -350,3 +389,137 @@ func webhooks() []webhookwrapper.WebhookRegistrator {
 		},
 	}
 }
+
+func setupClusterClientAndSetupCA(ctx context.Context, controllerWebhookService string, deploymentsNamespace string, validatingWebhooks string, mutatingWebhooks string, developmentMode bool) ([]byte, []byte, bool, error) {
+	// This client is not created inside the setupCA function just to
+	// facilitate the testing
+	clusterClient, err := client.New(ctrl.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		setupLog.Error(err, "unable to setup client")
+		return []byte{}, []byte{}, false, fmt.Errorf("failed to initialize the cluster client: %s", err.Error())
+	}
+	return setupCA(ctx, clusterClient, controllerWebhookService, deploymentsNamespace, strings.Split(strings.TrimSpace(validatingWebhooks), ","), strings.Split(strings.TrimSpace(mutatingWebhooks), ","), developmentMode)
+}
+
+// setupCA is a function used to initialize the root CA, the certificate used
+// by the controller to serve the webhook server.
+//
+// ctx is the context used with the clusterClient to get and patch cluster
+// information  resources.
+//
+// controllerWebhookService is the service name used to access the webhook server
+// available by the controller. This is necessary to properly generate the
+// certificate
+//
+// deploymentsNamespace is the namespace where the controller is deployed
+//
+// validatingWebhooks are the ValidatingWebhookConfiguration names which should
+// be updated when the root CA is initialized
+//
+// mutatingWebhooks are the MutatingWebhookConfiguration names which should
+// be updated when the root CA is initialized
+//
+// developmentMode is a flag to tell if the controller is running on
+// development mode or not
+func setupCA(ctx context.Context, clusterClient client.Client, controllerWebhookService string, deploymentsNamespace string, validatingWebhooks []string, mutatingWebhooks []string, developmentMode bool) ([]byte, []byte, bool, error) {
+	caSecret, caInitialized, err := admission.FetchOrInitializeCARootSecret(ctx, clusterClient, deploymentsNamespace, admissionregistration.GenerateCA, admissionregistration.PemEncodeCertificate)
+	if err != nil {
+		return []byte{}, []byte{}, false, fmt.Errorf("failed to fetch or initialize root CA certificate secret: %s", err.Error())
+	}
+	if caInitialized {
+		if err = admission.ReconcileSecret(ctx, clusterClient, caSecret); err != nil {
+			return []byte{}, []byte{}, false, fmt.Errorf("failed to reconcile root CA certificate secret: %s", err.Error())
+		}
+	}
+
+	certificate, privateKey, err := admission.ExtractCertificateData(caSecret)
+	if err != nil {
+		return []byte{}, []byte{}, false, fmt.Errorf("failed to initialize root CA certificate")
+	}
+
+	controllerSecret, initialized, err := admission.FetchOrInitializeCertificate(ctx, clusterClient, controllerWebhookService, deploymentsNamespace, constants.ControllerCertificateSecretName, caSecret, admissionregistration.GenerateCert)
```

Comment:
I think the code below this line should be moved to another function. The name of the this function is `setupCA`, hence it should only ensure the root CA is properly created.